### PR TITLE
feature: switch to two_factor for 2FA, supporting both SMS and authenticator/authy

### PR DESCRIPTION
**Pull request by liewegas**
- switch to two_factor
- support both SMS (via twilio) and authenticator/authy
- remove old multifactor code

TODO?
- [ ] remove old multifactor tables
- [x] clean up templates, esp for login/logout pages

### CODE REVIEW

File: scripts/remote_run.sh [hunk 1]
```diff
@@ -22,7 +22,6 @@ export REDIS_URL=$(aws ssm get-parameter --region $REGION --with-decryption --na
 export SECRET_KEY=$(aws ssm get-parameter --region $REGION --with-decryption --name turnout.$ENVIRONMENT.secret_key | jq '.Parameter["Value"]' -r)
 export ALLOWED_HOSTS=$(aws ssm get-parameter --region $REGION --with-decryption --name turnout.$ENVIRONMENT.allowed_hosts | jq '.Parameter["Value"]' -r)
 export SENTRY_DSN=$(aws ssm get-parameter --region $REGION --with-decryption --name turnout.$ENVIRONMENT.sentry_dsn | jq '.Parameter["Value"]' -r)
-export MULTIFACTOR_ISSUER=$(aws ssm get-parameter --region $REGION --with-decryption --name turnout.$ENVIRONMENT.multifactor_issuer | jq '.Parameter["Value"]' -r)
```
[the remark this answers is not in the document]
Comment (liewegas) [reply scripts/remote_run.sh:25]: I should do this as a follow-up, after this is deployed to production, right?

**Comment by liewegas** [comment]
> This looks great!
> 
> I added a few notes about configuring the new env vars in dev/staging/prod -- let me know if you want to pair on that. I simplified that process last week so now you just create the parameters in AWS SSM, and add them to https://github.com/vote/turnout/blob/master/ecs/template/include/secrets.libsonnet

Done

> I do think we want to clean up the UI a bit. Couple things in particular:

I fixed most of this via templates, but a few things come from the two_factor code itself.

> * We can remove the "cancel" link because we require 2fa -- it just takes you back to the "you must enable 2fa" screen

Dropped the cancel button throughout.
 
> * "You are about to take your account security to the next level. Follow the steps in this wizard to enable two-factor authentication." -> can we just remove this step? Or at lease rephrase to "Please follow the steps to enable two-factor authentication."

Can't remove the step, but adjusted the language.

> * Option select screen -> we should get rid of the bullet points, change "Token generator" to "Two-Factor App (Google Authenticator)"

Doesn't seem to be something I can change via the template

> * SMS 2fa token option -> if I get to the screen where I have to enter the token to confirm setup, and then I want to go back to switch to app-based 2fa, I can't click "Back" -- it shows a validation error on the token input. I have to enter something into the token box (just anything at all) to be able to use the back button.

Same.  This could probably be done with CSS but that's not something I've done before :(

> * "We sent you a text message, please enter the tokens we sent." -> "We have sent you a text message with a token. Please enter that token to confirm your phone number."
> * Should make the links/buttons on the Account Security page look more like links or buttons. Same with the links/button on the login page (to choose text message / backup token)

Fixed

> * Can we just get rid of the "Disable Two-Factor Authentication" option? We don't really let you disable it.

Disabling + reconfiguring is needed to change the primary method (e.g., text -> token generator).  I adjusted the language to make this clear.

File: scripts/remote_run.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (benweissmann) [reply scripts/remote_run.sh:26]: Yeah, I think it pulls the ECS configs from master so you should remove the old env var in a follow-up but you should add the new env var in this PR. It looks like something got messed up when generating the .task.json files -- in the PR, those files have been truncated.

[the remark this answers is not in the document]
Comment (liewegas) [reply scripts/remote_run.sh]: updated!

**Comment by benweissmann** [comment]
I pushed a couple of small style tweaks but I think there's still a bit more work to do -- I think especially the login page is pretty confusing and that's something people are going to see over and over.

I tried dropping in a `login.html` file but it didn't seem to override the login template -- is there something else I need to do to register a template override?

A couple things I think we should do here:

- Override as many of the templates as we can -- could you take a look at overriding `login.html` and any other files we can possibly override?

- I think "token generator" is going to be really confusing and most of our users won't know what that is -- can we override the English translation file? It looks like they're using an i18n package for all of the strings so if we can override that translation file, that should give us complete control over the copy.

- I think the only big flow issue is that if you choose text message as your primary authentication mechanism, when you reach the end of the wizard it says "However, it might happen that you don't have access to your primary token device. To enable account recovery, add a phone number." -- and prompts you to add a second phone. Is there any way to get rid of that? Or have it prompt you download backup codes instead of adding a second phone? I think the vast majority of our users are going to use SMS, so I think it's worth trying to clean up this setup flow.

- Once you've gotten those couple things working, I can take a UI/UX/CSS pass over the whole thing and then take some screenshots and send them over to Debra for approval.

**Comment by liewegas** [comment]
Okay, all of the templates are now there, and I made a pass over the language in the whole thing and I think it's in pretty good shape.  'Token generator' is defined on the page where it's used, so I don't think i18n changes are needed, and the buttons are adjusted so that the next step is always big and red.  The setup_complete step has buttons now to add a backup number or one-time codes, and less confusing copy.

The remaining rough edges are the bullet points in the formatting for the method selection, and the useless steps in the wizard flow.

Note that login.html is at ``app/manage/templates/management/auth/login.html``, not in ``two_factor``.

**Pull request merged**